cdc/sink/codec/canal_test.go

```diff
@@ -451,12 +451,19 @@ var testColumnsTable = []*testColumnTuple{
 
 func (s *canalEntrySuite) TestGetMySQLTypeAndJavaSQLType(c *check.C) {
 	defer testleak.AfterTest(c)()
+	canalEntryBuilder := NewCanalEntryBuilder()
 	for _, item := range testColumnsTable {
 		obtainedMySQLType := getMySQLType(item.column)
 		c.Assert(obtainedMySQLType, check.Equals, item.expectedMySQLType)
 
 		obtainedJavaSQLType, err := getJavaSQLType(item.column, obtainedMySQLType)
 		c.Assert(err, check.IsNil)
 		c.Assert(obtainedJavaSQLType, check.Equals, item.expectedJavaSQLType)
+
+		if !item.column.Flag.IsBinary() {
```

Comment:
Why only for binary?

Reply:
For non-binary. Because the `expectedValue` is dummy for binary: https://github.com/pingcap/tiflow/blob/master/cdc/sink/codec/canal_test.go#L424.